### ISSUE TITLE
libvirt-coreos kube-addon.service failed

### ISSUE THREAD

**Issue by MCyprien** [GoogleCloudPlatform/kubernetes#7218]
Hi,
I am using Kubernetes 15.0 with kvm. I followed the getting started guide for libvirt-coreos : https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/libvirt-coreos.md
But when I do kube-up.sh not a single node is added to Kubernetes:
```
Starting cluster using provider: libvirt-coreos
... calling verify-prereqs
... calling kube-up
KUBE_MASTER_IP: 192.168.10.1
KUBE_MASTER: kubernetes-master
--2015-04-23 10:23:17--  http://alpha.release.core-os.net/amd64-usr/current/core
os_production_qemu_image.img.bz2
Resolving alpha.release.core-os.net (alpha.release.core-os.net)... 173.194.65.12
8, 2a00:1450:4013:c00::80
Connecting to alpha.release.core-os.net (alpha.release.core-os.net)|173.194.65.1
28|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 142383897 (136M) [application/octet-stream]
Server file no newer than local file ‘cluster/../cluster/libvirt-coreos/coreos_p
roduction_qemu_image.img.bz2’ -- not retrieving.

Pool kubernetes refreshed

Network kubernetes_global created from cluster/../cluster/libvirt-coreos/network
_kubernetes_global.xml

Network kubernetes_pods created from cluster/../cluster/libvirt-coreos/network_k
ubernetes_pods.xml

Vol kubernetes-minion-1.img created

Pool kubernetes refreshed

Domain kubernetes-minion-1 created from /tmp/tmp.pJXbGPKXfV

Vol kubernetes-minion-2.img created

Pool kubernetes refreshed

Domain kubernetes-minion-2 created from /tmp/tmp.085mHBEZT8

Vol kubernetes-minion-3.img created
Pool kubernetes refreshed

Domain kubernetes-minion-3 created from /tmp/tmp.1qkILoK4fw

Vol kubernetes-master.img created

Pool kubernetes refreshed

Domain kubernetes-master created from /tmp/tmp.dIho46QC3A

current-context: ""
Running: cluster/../cluster/libvirt-coreos/../../cluster/../cluster/libvirt-core
os/../../platforms/linux/amd64/kubectl --server=http://192.168.10.1:8080 config
set-cluster libvirt-coreos --server=http://192.168.10.1:8080
open /root/.kube/.kubeconfig: no such file or directory
current-context: ""
Running: cluster/../cluster/libvirt-coreos/../../cluster/../cluster/libvirt-core
os/../../platforms/linux/amd64/kubectl --server=http://192.168.10.1:8080 config
set-context libvirt-coreos --cluster=libvirt-coreos
open /root/.kube/.kubeconfig: no such file or directory
current-context: ""
Running: cluster/../cluster/libvirt-coreos/../../cluster/../cluster/libvirt-core
os/../../platforms/linux/amd64/kubectl --server=http://192.168.10.1:8080 config
use-context libvirt-coreos --cluster=libvirt-coreos
open /root/.kube/.kubeconfig: no such file or directory
Wait for cluster readiness
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
Nb ready minions: 0 / 3
```

When I ssh to the master I can see that kube-addons.service failed.
here is sytemctl status kube-addons.service :
```
● kube-addons.service - Kubernetes addons
   Loaded: loaded (/etc/systemd/system/kube-addons.service; disabled; vendor pre
set: disabled)
   Active: failed (Result: exit-code) since Thu 2015-04-23 08:23:52 UTC; 14min a
go
     Docs: https://github.com/GoogleCloudPlatform/kubernetes
  Process: 655 ExecStart=/opt/kubernetes/bin/kubectl create -f /opt/kubernetes/a
ddons (code=exited, status=1/FAILURE)
  Process: 647 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)
  Process: 569 ExecStartPre=/bin/bash -c while [[ "$(curl -s http://127.0.0.1:80
80/healthz)" != "ok" ]]; do sleep 1; done (code=exited, status=0/SUCCESS)
 Main PID: 655 (code=exited, status=1/FAILURE)

Apr 23 08:23:38 kubernetes-master systemd[1]: Starting Kubernetes addons...
Apr 23 08:23:52 kubernetes-master kubectl[655]: Error: [502:  (unhandled http status [Temporary Redirect] with body []) [0], 502:  (unhandled http status [Temporary Redirect] with body []) [0]]
Apr 23 08:23:52 kubernetes-master systemd[1]: kube-addons.service: main process exited, code=exited, status=1/FAILURE
Apr 23 08:23:52 kubernetes-master systemd[1]: Failed to start Kubernetes addons.
Apr 23 08:23:52 kubernetes-master systemd[1]: Unit kube-addons.service entered failed state.
Apr 23 08:23:52 kubernetes-master systemd[1]: kube-addons.service failed.
```
And the journalctl _PID=655 is not very helpful and display the same error.

If it's not a bug, can someone help me with what is wrong?

**Comment by lhuard1A** [comment]
The error message you get looks like the issue reported in #6756 
@MCyprien : Could you check if you have the same “502: (unhandled http status [Temporary Redirect] with body []) [0]” errors in the kube-apiserver with `journalctl -u kube-apiserver`

If so, could you check which version on etcd is running?
```
ps aux | grep etcd
<the_path_you_found>/etcd --version
```

If you have etcd 0.4.9, that’s the reason of your problem.
Basically, CoreOS starts with an etcd 0.4.9 and kubernetes uses a 2.0.4 client.

I sent a fix in order to have etcd 2 server inside CoreOS.
This fix is #6756 
This fix is in the master branch, but not on the v0.15.0 tag.

You should either try a version which contains that fix or try to apply the patch of that PR to your v0.15.0.

**Comment by MCyprien** [comment]
thank you! that was the problem!
